data-prepper-plugins/cloudwatch-metrics-source/src/main/java/org/opensearch/dataprepper/plugins/source/CloudwatchMetricsWorker.java

```diff
@@ -0,0 +1,198 @@
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ *  implements the CloudwatchMetricsWorker to read and metrics data message and push to buffer.
+ */
+/**
+ *  An implementation of cloudwatch metrics source  worker class to write the metric to Buffer
+ */
+public class CloudwatchMetricsWorker implements Runnable{
+
+    private static final Logger LOG = LoggerFactory.getLogger(CloudwatchMetricsWorker.class);
+
+    private static final int STANDARD_BACKOFF_MILLIS = 30_000;
+
+    private final BufferAccumulator<Record<Event>> bufferAccumulator;
+
+    private final CloudWatchClient cloudWatchClient;
+
+    private final CloudwatchMetricsSourceConfig cloudwatchMetricsSourceConfig;
+
+    private final Collection<Dimension> dimensionCollection;
+
+    private final PluginMetrics pluginMetrics;
+
+    private final Function<Map<String, Object>, List<PartitionIdentifier>> partitionCreationSupplier;
+
+    private final SourceCoordinator<CloudwatchSourceProgressState> sourceCoordinator;
+
+    public CloudwatchMetricsWorker(final CloudWatchClient cloudWatchClient,
+                                   final BufferAccumulator<Record<Event>> bufferAccumulator,
+                                   final CloudwatchMetricsSourceConfig cloudwatchMetricsSourceConfig,
+                                   final Collection<Dimension> dimensionCollection,
+                                   final PluginMetrics pluginMetrics,
+                                   final SourceCoordinator<CloudwatchSourceProgressState> sourceCoordinator) {
+        this.bufferAccumulator = bufferAccumulator;
+        this.cloudWatchClient = cloudWatchClient;
+        this.cloudwatchMetricsSourceConfig = cloudwatchMetricsSourceConfig;
+        this.dimensionCollection = dimensionCollection;
+        this.pluginMetrics = pluginMetrics;
+        this.sourceCoordinator = sourceCoordinator;
+        final List<String> namespaceMetricNames = cloudwatchMetricsSourceConfig.getNamespacesListConfig().stream()
+                .map(NamespacesListConfig::getNamespaceConfig).flatMap(namespaceConfig -> namespaceConfig.getMetricDataQueriesConfig()
+                        .stream().map(metricName -> namespaceConfig.getName() + "@" + metricName.getMetricsConfig().getName())).collect(Collectors.toList());
+        this.partitionCreationSupplier = new CloudwatchPartitionCreationSupplier(namespaceMetricNames);
+    }
+
+    void startProcessingCloudwatchMetrics(final int waitTimeMillis){
+        final Optional<SourcePartition<CloudwatchSourceProgressState>> objectToProcess = sourceCoordinator.getNextPartition(partitionCreationSupplier);
+
+        if (objectToProcess.isEmpty()) {
+            try {
+                Thread.sleep(waitTimeMillis);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            return;
+        }
+        final String nameSpaceName = objectToProcess.get().getPartitionKey().split("@")[0];
```

Comment:
Please split this value only once. Also, CloudWatch namespaces do not allow `@`, but they may be possible in a Metric Name. I'm not clear on this from the documentation.

Please only split on the first `@`. This will ensure that even if an `@` shows up in the metric name, it will not interfere with the name.